src/tests/test_transform.py

```diff
@@ -2,15 +2,115 @@
 # This file may be used under the terms of the GNU General Public License, version 2 or later.
 # For more details see: https://www.gnu.org/licenses/gpl-2.0.html
 
+from src.transform.datastructures import AddonVersion, NVDAVersion, VersionCompatibility
+from src.transform.transform import getLatestAddons, _isAddonCompatible
+from src.tests.generateData import MockAddon
 import unittest
 
+V_2020_1 = NVDAVersion(2020, 1)
+V_2020_2 = NVDAVersion(2020, 2)
+V_2020_3 = NVDAVersion(2020, 3)
+V_2021_1 = NVDAVersion(2021, 1)
+V_2021_1_1 = NVDAVersion(2021, 1, 1)
+V_2021_2 = NVDAVersion(2021, 2)
+V_2022_1 = NVDAVersion(2022, 1)
+nvdaVersion2020_2 = VersionCompatibility(V_2020_2, V_2020_2, V_2020_1)
+nvdaVersion2020_3 = VersionCompatibility(V_2020_3, V_2020_3, V_2020_1)
+nvdaVersion2021_1 = VersionCompatibility(V_2021_1, V_2021_1, V_2021_1)
+nvdaVersion2021_1_1 = VersionCompatibility(V_2021_1_1, V_2021_1, V_2021_1)
+nvdaVersion2022_1 = VersionCompatibility(V_2022_1, V_2022_1, V_2022_1)
 
-class Test_testExample(unittest.TestCase):
-	def test_example_true(self):
-		self.assertTrue(True)
 
-	def test_example_pass(self):
-		pass
+class Test_isAddonCompatible(unittest.TestCase):
+	def test_valid_with_api(self):
+		"""Confirm an addon is compatible with a fully tested API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertTrue(_isAddonCompatible(addon, nvdaVersion2020_2))
 
-	# def test_example_fail(self):
-	# 	self.assertTrue(False, "Example test failure")
+	def test_valid_with_backwards_compatible_api(self):
+		"""Confirm an addon is compatible with a backwards compatible API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertTrue(_isAddonCompatible(addon, nvdaVersion2020_3))
+
+	def test_not_valid_because_breaking_api(self):
+		"""Confirm an addon is not compatible with a breaking API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2020_1
+		addon.lastTestedVersion = V_2020_2
+		self.assertFalse(_isAddonCompatible(addon, nvdaVersion2021_1))
+
+	def test_not_valid_because_old_api(self):
+		"""Confirm an addon is not compatible with an old API"""
+		addon = MockAddon()
+		addon.minNVDAVersion = V_2021_1
+		addon.lastTestedVersion = V_2021_2
+		self.assertFalse(_isAddonCompatible(addon, nvdaVersion2020_3))
+
+
+class Test_getLatestAddons(unittest.TestCase):
+	def test_beta_stable(self):
+		"""Ensure addons are added to the correct channels"""
```

Comment:
I'm not really sure what this test is showing. It the two versions have different values (other than stable / beta), but don't no versions exceed boundary conditions.

Reply:
I've re-written this to make the purpose clearer.
I'm not sure what boundary conditions you mean?

Reply:
Sorry, I'm not sure how I mangled that comment so much. I think I meant to point out that despite the different values, none of the different values cross boundaries which affect the behavior.

An example of testing either side of some boundary:
Setup two addon releases, older addon version that supports version 1 and 2 and a newer addon version that supports version 2. You could show that for version 1 == older addon release, for version 2 == newer addon

This code has a few of these boundaries. Look at the logic in the `_isAddonCompatible` and `_isAddonNewer` functions.